botocore/exceptions.py

```diff
@@ -286,6 +286,10 @@ class ClientError(Exception):
         'operation: {error_message}')
 
     def __init__(self, error_response, operation_name):
+        if 'Errors' in error_response and isinstance(error_response['Errors'],
```

Comment:
Pretty sure this is a bug in our error response parser.  We shouldn't be working around it in the ClientError `__init__`, we should just fix it at the source to prevent an invalid `error_response` from being created in the first place.